jit-format crash: Cannot create a file when that file already exists

```
D:\gh\coreclr3>jit-format --verbose -a x86 -o Windows -b Debug -c d:\gh\coreclr3
Formatting jit directory.
Formatting dll project.
Neither compile_commands.json exists, nor is there a build log. Running CMake to generate compile_commands.json.
Starting Build at 16:36:01.27
BUILD: Commencing CoreCLR Repo build
BUILD: Checking prerequisites
BUILD: Using environment: "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\\VsDevCmd.bat"
Tools are already initialized.
Running: d:\gh\coreclr3\Tools\dotnetcli\dotnet.exe d:\gh\coreclr3\Tools\run.exe d:\gh\coreclr3\config.json build -Project=d:\gh\coreclr3\build.pr
oj -generateHeaderWindows -NativeVersionHeaderFile="d:\gh\coreclr3\bin\obj\_version.h" -BuildOS=Windows_NT -BuildType=Debug -BuildArch=x86
Running: d:\gh\coreclr3\Tools\msbuild.cmd /nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false  d:\gh\coreclr3\build.proj /p:__B
uildType=Debug /p:__BuildArch=x86 /p:__BuildOS=Windows_NT  /t:GenerateVersionHeader /p:GenerateVersionHeader=true /p:NativeVersionHeaderFile=d:\g
h\coreclr3\bin\obj\_version.h
Command execution succeeded.
BUILD: Commencing build of native components for Windows_NT.x86.Debug
BUILD: Using environment: "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\\..\..\VC\vcvarsall.bat" x86
BUILD: Regenerating the Visual Studio solution
-- VS_PLATFORM_TOOLSET is
-- VS_PLATFORM_NAME is
-- Configuring done
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_RC_CREATE_SHARED_LIBRARY
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_RC_CREATE_SHARED_LIBRARY
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_RC_CREATE_SHARED_LIBRARY
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_RC_CREATE_SHARED_LIBRARY
-- Generating done
-- Build files have been written to: D:/gh/coreclr3/bin/nmakeobj/Windows_NT.x86.Debug
BUILD: Repo successfully built.  Finished at 16:36:13.10
BUILD: Product binaries are available at d:\gh\coreclr3\bin\Product\Windows_NT.x86.Debug

Unhandled Exception: System.IO.IOException: Cannot create a file when that file already exists
   at System.IO.Win32FileSystem.MoveFile(String sourceFullPath, String destFullPath)
   at System.IO.File.Move(String sourceFileName, String destFileName)
   at ManagedCodeGen.jitformat.Main(String[] args)
```

--- a/src/jit-format/jit-format.cs
+++ b/src/jit-format/jit-format.cs
@@ -179,6 +179,16 @@ namespace ManagedCodeGen
                 {
                     _syntaxResult.ReportError("Specify --coreclr");
                 }
+                else if (!Directory.Exists(_rootPath))
+                {
+                    // If _rootPath doesn't exist, it is an invalid path
+                    _syntaxResult.ReportError("Invalid path to coreclr directory. Specify with --coreclr");
+                }
+                else if (!File.Exists(Path.Combine(_rootPath, "build.cmd")) || !File.Exists(Path.Combine(_rootPath, "build.sh")))
+                {
+                    // If _rootPath\build.cmd or _rootPath\build.sh do not exist, it is an invalid path to a coreclr repo
+                    _syntaxResult.ReportError("Invalid path to coreclr directory. Specify with --coreclr");
+                }
 
                 // Check that we can find compile_commands.json on windows
                 if (_os == "Windows_NT")
@@ -412,7 +422,19 @@ namespace ManagedCodeGen
                 {
                     // Create the compile_commands directory. If it already exists, CreateDirectory will do nothing.
                     Directory.CreateDirectory(newCompileCommandsDir);
+
                     // Move original compile_commands file on non-windows
+                    try
+                    {
+                        // Delete newCompileCommands if it exists. If it does not exist, no exception is thrown
+                        File.Delete(newCompileCommands);
+                    }
+                    catch (DirectoryNotFoundException dirNotFound)
+                    {
+                        Console.WriteLine("Error deleting {0}", newCompileCommands);
+                        Console.WriteLine(dirNotFound.Message);
+                    }
+
                     File.Move(config.CompileCommands, newCompileCommands);
                 }
 

